lib/dynflow/execution_plan/steps/abstract.rb

```diff
@@ -112,8 +112,8 @@ def progress_weight
 
       # @return [Action] in presentation mode, intended for retrieving: progress information,
       # details, human outputs, etc.
-      def action(execution_plan)
-        world.persistence.load_action_for_presentation(execution_plan, action_id)
+      def action(execution_plan, with_step = false)
+        world.persistence.load_action_for_presentation(execution_plan, action_id, with_step ? self : nil)
```

Comment:
If you can check the usages from other places to make sure we don't break somethink, and make it a new default, I think it's better to have one option: the `with_step` option doesn't make much sense to me right now.